src/architecture/transactions.md

```diff
@@ -220,30 +216,49 @@ via [an appropriate transaction endpoint](services.md#transactions).
 
 !!! note
     As transactions use universally verified cryptography (digital signatures)
-    for authentication, a transaction theoretically can be submitted to the network
-    by anyone aware of the transaction. There is no intrinsic upper bound on
-    the transaction lifetime, either.
+    for authentication, a transaction theoretically can be submitted to the
+    network by anyone aware of the transaction. There is no intrinsic upper
+    bound on the transaction lifetime, either.
 
 !!! tip
     From the point of view of a light client, transaction execution is
     asynchronous; full nodes do not return an execution status synchronously
-    in a response to a client’s request. To determine transaction status,
-    you may poll the transaction status using [read requests](services.md#read-requests)
-    defined in the corresponding service or the blockchain explorer.
-    If a transaction is valid (i.e., its `verify` returns `true`), it’s expected
-    to be committed in a matter of seconds.
+    in response to the request of the client. To determine the transaction status,
+    you may poll the transaction status using
+    [read requests](services.md#read-requests) defined in the corresponding
+    service or the blockchain explorer. If a transaction is valid, it is
+    expected to be committed in a matter of seconds.
 
 ### 3. Verification
 
 After a transaction is received by a full node, it is looked up
 among committed transactions, using the transaction hash as the unique
-identifier. If a transaction has been committed previously, it is
+identifier. If the transaction has been committed previously, it is
 discarded, and the following steps are skipped.
 
 The transaction implementation is then looked up
 using the `(service_id, message_id)` type identifier.
-The `verify` method of the implementation is invoked to check the internal
-consistency of the transaction.
+
+If the transaction is absent in the blockchain, the `raw_from_buffer` method
+then checks that the byte array constituting the
+transaction message contains the author’s public key and signature and that the
+signature corresponds to the indicated public key.
+
+Next, if `raw_from_buffer` is successful, the byte array is converted into a
+`SignedMessage` structure. This structure contains all the fields of the
+message in the deserialized form, except for the message payload which is still
+presented as a byte array. This structure helps us avoid verifying the
+signature repeatedly during the next steps, as every `SignedMessage` is
+guaranteed to contain the correct signature.
+
+The `SignedMessage` is then passed to the `deserialize` method, which can
```

Comment:
This is not wrong, but it may cause confusion. 
We use `Signed<T>` structure almost everywhere in our code and this structure is what "contains both the `SignedMessage` and the deserialized payload." 
but also we have `Message` enum that is enum of `Signed<T>`

Reply:
I've made some changes to this and the following paragraph, please have a look whether it makes sense now.